src/backend/gl/src/queue.rs

```diff
@@ -616,19 +616,44 @@ impl Queue {
             com::Command::UnbindAttribute(ref attribute) => unsafe {
                 self.share.context.DisableVertexAttribArray(attribute.location);
             }*/
-            com::Command::CopyBufferToBuffer(src, dst, ref r) => unsafe {
+            com::Command::CopyBufferToBuffer(src, dst, _src_target, dst_target, ref r) => unsafe {
+                let is_index_buffer_only_element_dst =
+                    !self.share.private_caps.index_buffer_role_change
+                        && dst_target == glow::ELEMENT_ARRAY_BUFFER;
+
+                let copy_src_target = glow::COPY_READ_BUFFER;
+                let copy_dst_target = if is_index_buffer_only_element_dst {
```

Comment:
why not just use the `dst_target` directly?

Reply:
Moved comment higher 

`// WebGL not allowed to copy data from other targets to element buffer and can't copy element data to other buffers`